### ISSUE TITLE
DNU with super as receiver: debugger shows incorrect class name

### ISSUE THREAD

**Issue by ClotildeToullec** [guillep/pharo-core#10945]
**Bug description**
When there is a DNU and the receiver is `super`, the error printed in debugger is incorrect. Instead of showing the class of the object, it shows the superclass, i.e. the class where the lookup started.

**To Reproduce**
Small example:
- Class1 does not define `#thisMethod`.
- Class2 inherits from Class1 and defines: 
```Smalltalk
thisMethod
    ^ super thisMethod
```
- When executing `Class2 new thisMethod`, the DNU error message is:
'Instance of **Class1** did not understand #thisMethod'

**Expected behavior**
The DNU error message should be:
'Instance of **Class2** did not understand #thisMethod'

**Screenshots**
![Capture d’écran 2022-02-23 à 20 22 47](https://user-images.githubusercontent.com/39184695/155392340-5cbb213e-ff24-4cef-ab94-843c229807ea.png)

**Version information:**
 - Pharo 10

**Comment by MarcusDenker** [comment]
Hmm... but is this not correct? Class2 implements #thisMethod, thus it understand it:

```
Class2 canUnderstand:  #thisMethod
```
 returns true.

The DNU happens because of the super (which starts lookup at Class1).

```
Class1 canUnderstand:  #thisMethod

```

==> false

**Comment by ClotildeToullec** [comment]
Yes, that's correct, but the object is an instance of Class2, not of Class1.
The error message is confusing when debugging.
Example of misunderstanding: when the superclass is abstract: you will see 'Instance of AbstractSmthg ...' and start wondering how you ended up with an instance of an abstract class.

**Comment by MarcusDenker** [comment]
I think in the past the label was not using the "Instance of"... label, but just the class name. (before Pharo9?)

**Comment by MarcusDenker** [comment]
MessageNotUnderstood>>#description is where it is defined.

**Comment by MarcusDenker** [comment]
This is the PR that changed it: https://github.com/pharo-project/pharo/pull/5361

**Comment by MarcusDenker** [comment]
No, that one: https://github.com/pharo-project/pharo/issues/9907

**Comment by guillep** [comment]
#9907 is just a bug entry, not a PR. Maybe PR #5361 is related, but it only adds the `#` to the string. Not sure how this would have broken it (or maybe it was broken before of that)

**Comment by guillep** [comment]
Even with the change in #9982, the code change is:

https://github.com/pharo-project/pharo/pull/9982/files

```smalltalk
- ifFalse: [message lookupClass printString, '>> #', message selector asString]]]
+ ifFalse: ['Message not understood: ', message lookupClass printString, ' >> #', message selector asString]]]
```

So it was always using `message lookupClass`.

<blockquote><img src="https://avatars.githubusercontent.com/u/5469544?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/pharo-project/pharo/pull/9982">Fix #9907: improes DoesNotUnderstand exception's messageText by gypsydave5 · Pull Request #9982 · pharo-project/pharo</a></strong></div><div>Thrown a test in for good measure</div></blockquote>

**Comment by maxwills** [comment]
Maybe just completely denoting the method could be a clearer notation (for super lookup cases).
'Instance of Class2 did not understand Class1>>#thisMethod'
Although the phrasing is still weird, since Class1>>#thisMethod is a method and not a message selector.
